### ISSUE TITLE
Vulkan compile errors on mingw

### ISSUE THREAD

**Issue by fire** [godotengine/godot#31096]
**Godot version:**

1037fa4

**OS/device including version:**
compiling on docker / fedora 30

**Issue description:**

Is WinSock2 spelt wrong?

```bash
In file included from thirdparty/vulkan/loader/dev_ext_trampoline.c:22:
thirdparty/vulkan/loader/vk_loader_platform.h:28:10: fatal error: WinSock2.h: No such file or directory
 #include <WinSock2.h>
          ^~~~~~~~~~~~
compilation terminated.
scons: *** [thirdparty/vulkan/loader/dev_ext_trampoline.windows.opt.tools.64.o] Error 1
In file included from thirdparty/vulkan/loader/loader.h:33,
                 from thirdparty/vulkan/loader/asm_offset.c:24:
thirdparty/vulkan/loader/vk_loader_platform.h:28:10: fatal error: WinSock2.h: No such file or directory
 #include <WinSock2.h>
          ^~~~~~~~~~~~
compilation terminated.
scons: *** [thirdparty/vulkan/loader/asm_offset.windows.opt.tools.64.o] Error 1
drivers/wasapi/audio_driver_wasapi.cpp: In member function 'virtual ULONG CMMNotificationClient::Release()':
drivers/wasapi/audio_driver_wasapi.cpp:93:11: warning: deleting object of polymorphic class type 'CMMNotificationClient' which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
    delete this;
```

**Steps to reproduce:**

```scons platform=windows target=release_debug use_static_cpp=yes -j`nproc` use_lto=no use_llvm=yes gdnative_wrapper=yes deprecated=no```

**Comment by fire** [comment]
@bruvzg Ping

**Comment by akien-mga** [comment]
> Is WinSock2 spelt wrong?

You should be able to confirm this hypothesis easily :)

MinGW headers are lowercase indeed. Try to change it and see if it builds.

**Comment by fire** [comment]
Got my cicd system running again so it'll take a while to get through the backlog.

**Comment by akien-mga** [comment]
Oh I thought you had direct access to that host. I'll work on a fix then, I have mingw set up here on Linux.

I'll add some other fixes for comments I made post-merge in #29993.

**Comment by akien-mga** [comment]
Fixed by #31100.

**Comment by fire** [comment]
Thanks.
